src/vm/functions/database.rs

```diff
@@ -79,14 +102,26 @@ pub fn special_fetch_entry(args: &[SymbolicExpression],
         Explicit => eval(&args[1], env, &context)?
     };
 
-    env.global_context.database.fetch_entry(&env.contract_context.contract_identifier, map_name, &key)
+    let contract = &env.contract_context.contract_identifier;
+
+    // optimization todo: db metadata like this should just get stored
+    //   in the contract object, so that it gets loaded in when the contract
+    //   is loaded from the db.
+    let data_types = env.global_context.database.load_map(contract, map_name)?;
+    runtime_cost!(cost_functions::FETCH_ENTRY, env,
+                  cmp::max(data_types.value_type.size(),
+                           data_types.key_type.size()))?;
```

Comment:
Why is this `max`? Don't we want the sum of the two sizes?

Reply:
Yeah -- this should be a sum. I'm not sure why I thought this should be a max().